roofit/roofitcore/src/RooFactoryWSTool.cxx

```diff
@@ -241,13 +241,12 @@ namespace {
   static bool isValidEnumValue(const char* typeName, const char* value) {
     // Returns true if given type is an enum
 
-    // FIXME: We don't have any guarantee that typeName is less than 256 bytes
-    // Chop type name into class name and enum name
-    char buf[256];
-    strlcpy(buf,typeName,256);
-    char* className = strtok(buf,":");
-
-    return (strcmp(TEnum::GetEnum(className)->GetName(), value) == 0);
+    auto enumType = TEnum::GetEnum(typeName);
+    if (enumType) {
```

Comment:
That's too many braces for my eyes: what about `return (enumType) : enumType->GetConstant(value) ? false;`

Reply:
sure why not ...